File: src/main/java/itmo/is/project/dto/security/RegistrationRequest.java
package itmo.is.project.dto.security;

import com.fasterxml.jackson.annotation.JsonProperty;

public record RegistrationRequest(
        @JsonProperty(value = "username", required = true)
        String username,

        @JsonProperty(value = "password", required = true)
        String password
) {
}

File: src/main/kotlin/itmo/isproject/dto/security/RegistrationRequest.kt
package itmo.isproject.dto.security

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

data class RegistrationRequest(

    @Schema(example = "killreal777")
    @JsonProperty(value = "username", required = true)
    val username: String,

    @Schema(example = "qwerty12345")
    @JsonProperty(value = "password", required = true)
    val password: String
)
